firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -78,10 +85,18 @@ void homeAxis(uint8_t motor[])
   // and RIGHT arbitrarily while tuning end stop for both x and y axes.
 
   // Sets scale and direction for motor and current position
-  // Moving motor towards home for rough estimate
-  digitalWrite(motor[DIR_PIN], LEFT);
+  // Moving motor towards max or 0 for rough estimate
+  if (alignmentCode == POS_EXTREME)
+  {
+    digitalWrite(motor[DIR_PIN], RIGHT);
+    eighthStepsPerPulse = POS_EIGHTH_STEPS_PER_WHOLE_STEP;
+  }
+  else
+  {
+    digitalWrite(motor[DIR_PIN], LEFT);
+    eighthStepsPerPulse = NEG_EIGHTH_STEPS_PER_WHOLE_STEP;
+  }
   setScale(motor, WHOLE_STEPS);
-  eighthStepsPerPulse = NEG_EIGHTH_STEPS_PER_WHOLE_STEP;
   while (digitalRead(motor[ENDSTOP_PIN]) == LOW)
```

Comment:
I should have caught this in the last PR, but since we added an additional element to the `motor` array type, we need to update the preprocessors defining the endstop pins. There should be two defs at line 75 in `chessboard.ino`:
```c
#define ZERO_ENDSTOP_PIN 4
#define MAX_ENDSTOP_PIN 5
```
After updating the defs you'll need to update this function to use those new pins where appropriate.

Reply:
To update the function using the new endstop pins, should I just put the while loop inside the if and else, depending on which way the gantry is aligning too? I would put in the if to use MAX_ENDSTOP and the else would use ZERO_ENDSTOP

Reply:
Nashir can check me on this, but they need to be there cuz the digitalRead evaluates the state of the button each loop.